tracecat/actions/integrations/iam/ldap.py

```diff
@@ -0,0 +1,197 @@
+"""LDAP integration.
+
+Authentication method: Token
+
+Requires: A secret named `ldap` with the following keys:
+- `LDAP_BIND_DN`
+- `LDAP_BIND_PASS`
+
+"""
+
+import json
+import os
+from typing import Annotated, Any
+
+import ldap3
+
+from tracecat.registry import Field, RegistrySecret, registry
+
+ldap_secret = RegistrySecret(
+    name="ldap",
+    keys=["LDAP_BIND_DN", "LDAP_BIND_PASS"],
+)
+"""LDAP secret.
+
+- name: `ldap`
+- keys:
+    - `LDAP_BIND_DN`
+    - `LDAP_BIND_PASS`
+"""
+
+
+class LdapClient:
+    _ldap_server = None
```

Comment:
@mattdurant is there any reason why you've set these variables as class as opposed to instance attributes? So if it's declared here, these variables are scoped to the class and (may) introduce side effects across multiple instances.

For safety, it's better to define all attributes scoped at the instance level (i.e. under `def __init__`).

Reply:
@topher-lo I missed this; I initially built this integration as a separate script we used internally and rather than re-implement most of it I encapsulated the original script in a class and then refactored to match how other integrations are built. Corrected in latest commit.